### ISSUE TITLE
Not safe case (or false positive?) of `Style/RedundantLineContinuation` cop

### ISSUE THREAD

**Issue by r-plus** [rubocop/rubocop#15087]
### Before submitting

- [x] I have searched the existing issues and this hasn't been reported yet.
- [x] I am running the latest stable version of RuboCop.

### Issue type

False positive

### Cop name

Style/RedundantLineContinuation

### Steps to reproduce the problem

test code
```ruby
#!/usr/local/bin ruby

# frozen_string_literal: true

def a
  'a'
end

def foobar
  "longlong-string \
#{a}"
end

puts foobar
```

result before autocorrect.

```
$ ruby ./test.rb
longlong-string a
```

autocorrect.

```
$ bundle exec rubocop -a test.rb
Inspecting 1 file
C

Offenses:

test.rb:10:19: C: [Corrected] Layout/TrailingWhitespace: Trailing whitespace detected.
  "longlong-string
                  ^
test.rb:10:20: C: [Corrected] Style/RedundantLineContinuation: Redundant line continuation.
  "longlong-string \ ...
                   ^

1 file inspected, 2 offenses detected, 2 offenses corrected
```

autocorrected diff

```diff
 def foobar
-  "longlong-string \
+  "longlong-string
 #{a}"
 end
```

after autocorrect, result added line-break.
it will change the behavior.

```
$ ruby ./test.rb
longlong-string
a
```

### Expected behavior

not offense or somethings?

### Actual behavior

added line-break by simply remove the line-continuation backslach.

### RuboCop configuration

```yaml
AllCops:
  NewCops: enable
Metrics:
  Enabled: false
Layout/LineLength:
  Enabled: false
```

### RuboCop version

```shell
$ bundle exec rubocop -V
1.86.0 (using Parser 3.3.11.1, Prism 1.8.1, rubocop-ast 1.49.1, analyzing as Ruby 4.0, running on ruby 4.0.2) [arm64-darwin25]
```
